drivers/serial/uart_nrfx_uarte.c

```diff
@@ -931,6 +931,17 @@ static void rx_timeout(struct k_timer *timer)
 	int32_t len = data->async->rx_total_byte_cnt
 		    - data->async->rx_total_user_byte_cnt;
 
+	if (!hw_rx_counting_enabled(data) &&
+	    (len < 0)) {
+		/* Prevent too low value of rx_cnt.cnt which may occur due to
+		 * latencies in handling of the RXRDY interrupt. Because whole
+		 * buffer was filled we can be sure that rx_total_user_byte_cnt
+		 * is current total number of received bytes.
```

Comment:
The second sentence in this comment is a bit confusing in this context, as we don't handle a filled buffer here. I suggest to replace it with something like: "at this point, the number of received bytes is at least equal to what was reported to the user".

Reply:
Updated according to the suggestion.